tests/stdlib/tjson.nim

```diff
@@ -244,3 +244,43 @@ when defined(js): # xxx fixme
   doAssert x.kind == JInt
 else:
   doAssert x.kind == JString
+
+
+type
+  Foo = object
+    ii*: int
+    data*: JsonNode
+
+block:
+  const jt = """{"ii": 123, "data": ["some", "data"]}"""
+  let js = parseJson(jt)
+  discard js.to(Foo)
+
+block:
+  const jt = """{"ii": 123}"""
+  let js = parseJson(jt)
+  doAssertRaises(KeyError):
+    echo js.to(Foo)
+
```

Comment:
```suggestion
block: # bug #15835
  type
    Foo = object
      ii*: int
      data*: JsonNode

  block:
    const jt = """{"ii": 123, "data": ["some", "data"]}"""
    let js = parseJson(jt)
    discard js.to(Foo)

  block:
    const jt = """{"ii": 123}"""
    let js = parseJson(jt)
    doAssertRaises(KeyError):
      echo js.to(Foo)

```

(@xflywind hence the need for `block: # bug #1234` even if test is under t1234.nim, it avoids forgetting this reference)

Reply:
make sense